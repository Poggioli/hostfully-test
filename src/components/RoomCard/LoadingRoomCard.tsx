import {
  Card
} from "../ui/card";
import { FC } from "react";
import { Skeleton } from "../ui/skeleton";


const LoadingRoomCard: FC = () => {
  return (
    <Card className="w-full">
      <Skeleton className="h-[200px] w-full" />
      <div className="p-3 flex flex-col gap-4">
        <Skeleton className="h-6 w-full max-w-[150px]" />
        <Skeleton className="h-14 w-full" />
        <Skeleton className="h-5 w-full max-w-[100px]" />

        <div className="flex justify-between gap-4">
          <Skeleton className="h-9 w-full max-w-[100px]" />
          <Skeleton className="h-9 w-full max-w-[100px]" />
        </div>
      </div>
    </Card>
  )
}

export { LoadingRoomCard };
