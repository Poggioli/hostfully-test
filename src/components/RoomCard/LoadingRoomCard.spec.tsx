import { render } from "@testing-library/react";
import { LoadingRoomCard } from "./LoadingRoomCard";

describe('LoadingRoomCard', () => {
  it(`GIVEN a LoadingRoomCard
      WHEN render
      THEN should render all props the right way`, () => {
    const { baseElement } = render(<LoadingRoomCard />);
    expect(baseElement).toMatchSnapshot();
  });
})