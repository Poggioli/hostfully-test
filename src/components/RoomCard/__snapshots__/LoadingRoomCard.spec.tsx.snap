// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingRoomCard GIVEN a LoadingRoomCard
      WHEN render
      THEN should render all props the right way 1`] = `
<body>
  <div>
    <div
      class="rounded-xl border bg-card text-card-foreground shadow w-full"
    >
      <div
        class="animate-pulse rounded-md bg-primary/10 h-[200px] w-full"
      />
      <div
        class="p-3 flex flex-col gap-4"
      >
        <div
          class="animate-pulse rounded-md bg-primary/10 h-6 w-full max-w-[150px]"
        />
        <div
          class="animate-pulse rounded-md bg-primary/10 h-14 w-full"
        />
        <div
          class="animate-pulse rounded-md bg-primary/10 h-5 w-full max-w-[100px]"
        />
        <div
          class="flex justify-between gap-4"
        >
          <div
            class="animate-pulse rounded-md bg-primary/10 h-9 w-full max-w-[100px]"
          />
          <div
            class="animate-pulse rounded-md bg-primary/10 h-9 w-full max-w-[100px]"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
